packages/core-interfaces/src/logging.ts

```diff
@@ -0,0 +1,161 @@
+import { HttpRequest } from './httpRequest';
+import { HttpResponse } from './httpResponse';
+/**
+ * Represents an interface for logging API requests and responses.
+ */
+export interface ApiLoggerInterface {
+  /**
+   * Logs the details of an HTTP request.
+   * @param request The HTTP request to log.
+   */
+  logRequest(request: HttpRequest): void;
+
+  /**
+   * Logs the details of an HTTP response.
+   * @param response The HTTP response to log.
+   */
+  logResponse(response: HttpResponse): void;
+}
+
+/**
+ * Represents an interface for a generic logger.
+ */
+export interface LoggerInterface {
+  /**
+   * Logs a message with a specified log level and additional parameters.
+   * @param level The log level of the message.
+   * @param message The message to log.
+   * @param params Additional parameters to include in the log message.
+   */
+  log(level: LogLevel, message: string, params: Record<string, any>): void;
+}
+
+/**
+ * Represents options for configuring logging behavior.
+ */
+export interface LoggingOptions {
+  /**
+   * The logger to use for logging messages.
+   */
+  logger: LoggerInterface;
+
+  /**
+   * The log level to determine which messages should be logged.
+   */
+  logLevel: LogLevel;
+
+  /**
+   * Options for logging HTTP requests.
+   */
+  logRequest: HttpRequestLoggingOptions;
+
+  /**
+   * Options for logging HTTP responses.
+   */
+  logResponse: HttpMessageLoggingOptions;
+
+  /**
+   * Indicates whether sensitive headers should be masked in logged messages.
+   */
+  maskSensitiveHeaders: boolean;
+}
+
+/**
+ * Represents options for configuring SDK logging behavior.
+ */
+export interface SdkLoggingOptions {
```

Comment:
I don't believe this interface is used anywhere in the code. Please remove it.

Reply:
yes I removed that one just now, its not needed anymore